src/utils/triton.h

```diff
@@ -0,0 +1,177 @@
+
+namespace triton { namespace backend { namespace dali {
+
+
+/**
+ * Converts TRITONSERVER_DataType to dali_data_type_t
+ */
+inline dali_data_type_t to_dali(TRITONSERVER_DataType t) {
+  assert(t >= 0 && t <= 13);
+  // Use the trick, that TRITONSERVER_DataType and dali_data_type_t
+  // types have more-or-less the same order, with few exceptions
+  if (t == 0)
+    return static_cast<dali_data_type_t>(-1);
+  if (t == 1)
+    return static_cast<dali_data_type_t>(11);
+  if (t == 13)
+    return static_cast<dali_data_type_t>(0);
+  return static_cast<dali_data_type_t>(t - 2);
+}
+
+/**
+ * Converts TRITONSERVER_MemoryType to DALI device_type_t
+ */
+inline device_type_t to_dali(TRITONSERVER_MemoryType t) {
+  switch (t) {
+    case TRITONSERVER_MEMORY_CPU:
+    case TRITONSERVER_MEMORY_CPU_PINNED:
+      return device_type_t::CPU;
+    case TRITONSERVER_MEMORY_GPU:
+      return device_type_t::GPU;
+    default:
+      throw std::invalid_argument("Unknown memory type");
+  }
+}
+
+/**
+ * Converts dali_data_type_t to TRITONSERVER_DataType
+ */
+inline TRITONSERVER_DataType to_triton(dali_data_type_t t) {
+  assert(-1 <= t && t <= 11);
+  // Use the trick, that TRITONSERVER_DataType and dali_data_type_t
+  // types have more-or-less the same order, with few exceptions
+  if (t == -1)
+    return static_cast<TRITONSERVER_DataType>(0);
+  if (t == 11)
+    return static_cast<TRITONSERVER_DataType>(1);
+  return static_cast<TRITONSERVER_DataType>(t + 2);
+}
+
+/**
+ * Converts device_type_t to TRITONSERVER_MemoryType
+ */
+inline TRITONSERVER_MemoryType to_triton(device_type_t dev) {
+  switch (dev) {
+    case CPU:
+      return TRITONSERVER_MEMORY_CPU_PINNED;
+    case GPU:
+      return TRITONSERVER_MEMORY_GPU;
+    default:
+      throw std::invalid_argument("Unknown memory type");
+  }
+}
+
+class TritonInput {
+ public:
+  TritonInput(TRITONBACKEND_Input *handle) : handle_(handle) {
+    TRITONSERVER_DataType input_datatype;
+    const int64_t *input_shape;
+    uint32_t input_dims_count;
+    uint64_t input_byte_size;
+    uint32_t input_buffer_count;
+    const char *name;
+    TRITON_CALL_GUARD(TRITONBACKEND_InputProperties(handle_, &name, &input_datatype, &input_shape,
+                                                    &input_dims_count, &byte_size_, &buffer_cnt_));
+    meta_.name = std::string(name);
+    meta_.type = to_dali(input_datatype);
+    auto batch_size = input_shape[0];
+    auto sample_shape = TensorShape<>(input_shape + 1, input_shape + input_dims_count);
+    auto shape = TensorListShape<>::make_uniform(batch_size, sample_shape);
+    meta_.shape = shape;
+  }
+
+  IOMeta Meta() const {
+    return meta_;
+  }
+
+  size_t ByteSize() const {
+    return byte_size_;
+  }
+
+  uint32_t BufferCount() const {
+    return buffer_cnt_;
+  }
+
+  /**
+   * Request an input buffer.
+   * idx  number index
+   * device_type_t preferred device type
+   * device_id preferred device id
```

Comment:
Don't you want to use doxygen convention?

Reply:
Fixed. (is it doxygen convention now?)